app/feed/page.tsx

```diff
@@ -1,94 +1,54 @@
-  const regex = /<([^>]+)>; rel="([^"]+)"/g;
-  const pageNums = {
-    first: undefined as string | undefined,
-    prev: undefined as string | undefined,
-    next: undefined as string | undefined,
-    last: undefined as string | undefined,
-  };
-
-  let match;
-  while ((match = regex.exec(text)) !== null) {
-    const page = new URL(match[1]).searchParams.get("page");
-    pageNums[match[2] as keyof typeof pageNums] = page
-      ? `/feed?page=${page}`
-      : undefined;
-  }
-
-  return pageNums;
-}
-
-export default function FeedPage({ searchParams }: Props) {
-  const [lastFetchedPage, setLastFetchedPage] = useState(1);
-  const [events, setEvents] = useState<Record<number, IGitHubEvent[]>>({});
-
-  useEffect(() => {
-    fetch(
-      `https://api.github.com/orgs/${env.NEXT_PUBLIC_GITHUB_ORG}/events?per_page=1000&page=${lastFetchedPage}`,
-    )
-      .then((res) => res.json())
-      .then((data: IGitHubEvent[]) => {
-        setEvents((existing) => ({
-          ...existing,
-          page: combineSimilarPushEvents(
-            data.filter(exludeBotEvents).filter(excludeBlacklistedEvents),
-          ),
-        }));
-
-        if (lastFetchedPage !== 1) {
-          const lastEvents = events[lastFetchedPage - 1];
-          scrollTo(`gh-event-${lastEvents[lastEvents.length - 1].id}`);
-        }
-      });
-  }, [events, lastFetchedPage]);
-
-  const allEvents = ([] as IGitHubEvent[]).concat(...Object.values(events));
-
+export default async function FeedPage({ searchParams }: Props) {
+  const events = await octokit.paginate(
+    "GET /orgs/{org}/events",
+    {
+      org: GITHUB_ORG,
+      per_page: 1000,
+    },
+    (response) => {
+      const data = response.data as IGitHubEvent[];
+      const filteredData = data
+        .filter(exludeBotEvents)
+        .filter(excludeBlacklistedEvents);
+      return filteredData;
+    },
+  );
   if (!Object.entries(events).length) {
     return <LoadingText text="Fetching latest events" />;
   }
-
   return (
     <div className="relative mx-auto my-8 flow-root max-w-4xl p-4">
       <h1 className="text-4xl text-primary-500 dark:text-white">Feed</h1>
       <ul role="list" className="mb-20 mt-10 flex flex-col gap-4 space-y-4">
-        {allEvents.map((e) => (
+        {events.map((e) => (
           <GitHubEvent key={e.id} event={e} />
         ))}
       </ul>
-      <div className="flex flex-row justify-center">
-        <span
-          className="cursor-pointer underline"
-          onClick={() => setLastFetchedPage((p) => p + 1)}
-        >
-          Show more
-        </span>
-      </div>
     </div>
   );
 }
 
-const exludeBotEvents = (event: IGitHubEvent) => {
-  return !event.actor.login.includes("bot");
-};
-
-const excludeBlacklistedEvents = (event: IGitHubEvent) => {
+const excludeBlacklistedEvents = (event: IGitHubEvent): boolean => {
```

Comment:
Why was this changed? Revert unrelated changes

<img width="2106" alt="image" src="https://github.com/coronasafe/leaderboard/assets/25143503/906152c7-df14-497c-97a7-9038b2b42139">